src/Fantomas.Core/Context.fs

```diff
@@ -905,41 +923,28 @@ let addParenIfAutoNln expr f =
     let expr = f expr
     expressionFitsOnRestOfLine expr (ifElse hasParenthesis (sepOpenT +> expr +> sepCloseT) expr)
 
-let autoIndentAndNlnExpressUnlessStroustrup (f: Expr -> Context -> Context) (e: Expr) (ctx: Context) =
+let autoIndentAndNlnExpressUnlessSameLinePreferred f (e: Expr) (ctx: Context) =
     let shouldUseStroustrup =
-        ctx.Config.ExperimentalStroustrupStyle
-        && e.IsStroustrupStyleExpr
-        && let node = Expr.Node e in
-           Seq.isEmpty node.ContentBefore
+        isStroustrupApplicable e.IsStroustrupStyleExpr (Expr.Node e) ctx
 
-    if shouldUseStroustrup then
+    if shouldUseStroustrup || isNamedComputationAndPreferSameLine e ctx then
         f e ctx
     else
         indentSepNlnUnindent (f e) ctx
 
-let autoIndentAndNlnTypeUnlessStroustrup (f: Type -> Context -> Context) (t: Type) (ctx: Context) =
+let autoIndentAndNlnTypeUnlessStroustrup f (t: Type) (ctx: Context) =
     let shouldUseStroustrup =
-        ctx.Config.ExperimentalStroustrupStyle
-        && t.IsStroustrupStyleType
-        && let node = Type.Node t in
-           Seq.isEmpty node.ContentBefore
+        isStroustrupApplicable t.IsStroustrupStyleType (Type.Node t) ctx
 
     if shouldUseStroustrup then
         f t ctx
     else
         autoIndentAndNlnIfExpressionExceedsPageWidth (f t) ctx
 
-let autoIndentAndNlnIfExpressionExceedsPageWidthUnlessStroustrup
-    (f: Expr -> Context -> Context)
-    (e: Expr)
-    (ctx: Context)
-    =
-    let isStroustrup =
-        ctx.Config.ExperimentalStroustrupStyle
-        && e.IsStroustrupStyleExpr
-        && Seq.isEmpty (Expr.Node e).ContentBefore
+let autoIndentAndNlnIfExpressionExceedsPageWidthUnlessSameLinePreferred f (e: Expr) (ctx: Context) =
+    let isStroustrup = isStroustrupApplicable e.IsStroustrupStyleExpr (Expr.Node e) ctx
 
-    if isStroustrup then
+    if isNamedComputationAndPreferSameLine e ctx || isStroustrup then
```

Comment:
If there is `TriviaBefore` in the `NamedComputation` that should be captured as well.
Example:
```fsharp
[<Test>]
let ``prefer computation expression name on same line, with trivia`` () =
    formatSourceString
        false
        """
let t =
    //
    task {
        let! thing = otherThing ()
        return 5
    }
"""
        config
    |> prepend newline
    |> should
        equal
        """
let t =
    //
    task {
        let! thing = otherThing ()
        return 5
    }
"""
```